File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        i,res,maxcount = 0,0,0
        count = {}
        for r in range(len(s)):
            count[s[r]] = 1 +  count.get(s[r],0)
            maxcount = max(maxcount,count[s[r]])
            while(r-i+1-maxcount>k):
                count[s[i]] -=1
                i +=1
            res = max(res,r-i+1)
            r +=1
        return res            
        

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.length();
        int res = 0, mf = 0;
        int i=0,j=0;
        int count[26] = {0};
        while(j<n)
        {
          count[s[j]-'A']++;
          mf = max(mf,count[s[j]-'A']);
          if(j-i+1 - mf > k){
            count[s[i]-'A']--;
            mf=0;
            i++;
          }
          else res = max(res,j-i+1);
          j++;
        }
        return res;
    }
};